aws/sdk/integration-tests/s3/tests/required-query-params.rs

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use aws_sdk_s3::operation::AbortMultipartUpload;
+use aws_sdk_s3::Region;
+use aws_smithy_http::operation::error::BuildError;
+
+#[tokio::test]
+async fn test_error_when_required_query_param_is_unset() {
```

Comment:
I think we should also have a Kotlin unit test for this query param validation functionality, since it applies more broadly than to just S3.